backend/api/routes/notifications.js

```diff
@@ -22,4 +23,31 @@ router.route("/").get(authenticateMiddleware, async (req, res) => {
   }
 });
 
+//will update notifcation status whether it be delivered, read, etc
+router.route("/:id").put(authenticateMiddleware, async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { status } = req.body;
+    const user_id = req.user.id;
+    const supabaseClient = req.supabase;
+
+    const { error } = await supabaseClient
+      .from("notifications")
+      .update({
+        status: status,
+        //if status for notif is that it was read then update read_at field will need this later for tracking when user is most active viewing notifs
+        read_at: status === "read" ? new Date().toISOString() : null,
```

Comment:
What are the status that you're planning to use? Instead of `read_at`, maybe something more generic like `updated_at` could be useful if you have something like "accepted", "deleted", etc. That way you know when the user is interacting with the notifications and not only reading them.